spec/statement.md

```diff
@@ -0,0 +1,181 @@
+</pre></x>
+
+1: This production is mandatory unless the previous `Statement` is a
+`LoudComment`, `SilentComment`, or ends in a `Block`.
+
+If a `WhitespaceComment` would be ambiguous with a `Statement` in the
+`ScssStatements` rule, parse it preferentially as a `Statement`.
+
+### IndentedStatements
+
+<x><pre>
+**IndentedStatements**  ::= (Statement [IndentSame])* Statement
+</pre></x>
+
+[IndentSame]: #indentation
+
+The `Statement` productions may not include newlines outside of `IndentSame`
+productions.
+
+### Stylesheet
+
+<x><pre>
+**Stylesheet**          ::= U+FEFF? ([ScssStatements] | [IndentedStatements])¹
+</pre></x>
+
+[ScssStatements]: #scssstatements
+[IndentedStatements]: #indentedstatements
+
+1: Only the production for the current syntax is valid.
+
+### Block
+
+<x><pre>
+**ScssBlock**      ::= '{' [ScssStatements] '}'
+**IndentedBlock**  ::= [IndentMore] [IndentedStatements]
+**Block**          ::= (ScssBlock | IndentedBlock)¹
+</pre></x>
+
+[IndentMore]: #indentation
+
+1: Only the production for the current syntax is valid.
+
+### Comments
+
+#### LoudComment
+
+<x><pre>
+**ScssLoudComment**          ::= '/\*' (.\*¹ | Interpolation)\* '\*/'
+**InterpolatedCommentText**  ::= (.\*² | Interpolation)\*
+**IndentedLoudChildren**     ::= (InterpolatedCommentText [IndentSame])\*
+&#32;                            InterpolatedCommentText
+**IndentedLoudComment**      ::= '/\*' InterpolatedCommentText
+&#32;                            ([IndentMore] IndentedLoudChildren)?
+**LoudComment**              ::= (ScssLoudComment | IndentedLoudComment)³
+</pre></x>
+
+1: This may not contain `#{` or `*/`.
+
+2: This may not contain `#{` or newlines.
+
+3: Only the production for the current syntax is valid.
+
+#### SilentComment
+
+<x><pre>
+**ScssSilentComment**          ::= '//' .\*¹
+**CommentText**                ::= .\*²
+**IndentedSilentChildren**     ::= (CommentText [IndentSame])\* CommentText
+**IndentedSilentComment**      ::= '//' CommentText
+&#32;                              ([IndentMore] IndentedSilentChildren)?
+**SilentComment**              ::= (ScssSilentComment | IndentedSilentComment)³
+</pre></x>
+
+1: This may not contain newlines.
+
+2: This may not contain newlines outside of [WhitespaceOnlyLine] productions.
+
+3: Only the production for the current syntax is valid.
+
+[WhitespaceOnlyLine]: #indentation
+
+#### WhitespaceComment
+
+<x><pre>
+**WhitespaceComment**              ::= ('//' .\*¹) | ('/\*' .\*² '\*/')
+</pre></x>
+
+1: This may not contain newlines.
+
+2: This may not contain `*/`. In the indented syntax, this may not contain
+   newlines.
+
+### Whitespace
+
+<x><pre>
+**LineBreak**  ::= CarriageReturn | LineFeed | FormFeed
+**Whitespace** ::= LineBreak¹ | Space | Tab | [WhitespaceComment]
+</pre></x>
+
+1: This is not allowed in the indented syntax.
+
+[WhitespaceComment]: #whitespacecomment
+
+### Indentation
+
+<x><pre>
+**WhitespaceOnlyLine**          ::= IndentSame? [Whitespace]\* [LineBreak]
```

Comment:
There's not really a reason to have `IndentSame?` here. The point is that empty lines can have any whitespace and have no bearing on the current indentation level.